### ISSUE TITLE
Can't call setState on a component that is not yet mounted

### ISSUE THREAD

**Issue by tomtom94** [benevbright/react-navigation-collapsible#13]
I am trying to change the default class state with setState from the const ExtraHeader but this is impossible.

**Warning: Can't call setState on a component that is not yet mounted. This is a no-op, but it might indicate a bug in your application. Instead, assign to `this.state` directly or define a `state = {};` class property with the desired state in the SearchEngine component.**

I even put the const inside my default class just to be sure, nothing works

In your example you define the const ExtraHeader like this below

```
const ExtraHeader = (/*{navigation}*/) => (
  <View style={{width: '100%', height: '100%', paddingHorizontal: 20, paddingVertical: 10}}>
    <View style={{backgroundColor: 'white', flex: 1, borderRadius: 15, justifyContent: 'center', alignItems: 'center'}}>
      <Text style={{color: 'gray'}}>Search Here</Text>
    </View>
  </View>
);
```

Have you ever tried to put an onPress in the Text balise.

```
<Text 
style={{color: 'gray'}} 
onPress={() => {
this.setState({ searchState: "coucou" });
}}>
Search Here
</Text>
```

This won't work. Do you have an alternative ?

**Comment by tomtom94** [comment]
I just took your example at this page https://github.com/benevbright/react-navigation-collapsible/blob/master/example/src/ExtraHeaderScreen.js

And I made some light modifications in order to put a setState on the search bar.
```

import React, { Component } from "react";
import { Text, FlatList, View, Animated, TouchableOpacity } from "react-native";

import { withCollapsible } from "react-navigation-collapsible";

const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);

class ExtraHeaderScreen extends Component {
  static navigationOptions = {
    title: "Extra Header"
  };

  constructor(props) {
    super(props);

    const data = [];
    for (let i = 0; i < 60; i++) {
      data.push(i);
    }

    this.state = {
      data: data,
      searchBar: "" // <---------- I wanna get the query of the search
    };
  }

  renderItem = ({ item }) => (
    <TouchableOpacity
      onPress={() => {
        this.props.navigation.navigate("DetailScreen");
      }}
      style={{
        width: "100%",
        height: 50,
        borderBottomColor: "#0002",
        borderBottomWidth: 0.5,
        paddingHorizontal: 20,
        justifyContent: "center"
      }}
    >
      <Text style={{ fontSize: 22 }}>{item}</Text>
    </TouchableOpacity>
  );

  render() {
    const { paddingHeight, scrollY, onScroll } = this.props.collapsible;

    return (
      <AnimatedFlatList
        style={{ flex: 1 }}
        data={this.state.data}
        renderItem={this.renderItem}
        keyExtractor={(item, index) => String(index)}
        contentContainerStyle={{ paddingTop: paddingHeight }}
        scrollIndicatorInsets={{ top: paddingHeight }}
        onScroll={onScroll}
        _mustAddThis={scrollY}
      />
    );
  }
}

const objExtraHeaderScreen = new ExtraHeaderScreen(); // <---------- In order to access the class above

const ExtraHeader = (/*{navigation}*/) => (
  <View
    style={{
      width: "100%",
      height: "100%",
      paddingHorizontal: 20,
      paddingVertical: 10
    }}
  >
    <View
      style={{
        backgroundColor: "white",
        flex: 1,
        borderRadius: 15,
        justifyContent: "center",
        alignItems: "center"
      }}
    >
      <Text
        style={{ color: "gray" }}
        onPress={() => {
          objExtraHeaderScreen.setState({ searchBar: "what's up guys ?" }); // <---------- The simple setState because it is necessary to put a setState if there is a search bar ;)
        }}
      >
        Search Here
      </Text>
    </View>
  </View>
);

const collapsibleParams = {
  extraHeader: ExtraHeader,
  extraHeaderStyle: {
    height: 50,
    backgroundColor: "#061"
  }
};

export default withCollapsible(ExtraHeaderScreen, collapsibleParams);
```

Warning: Can't call setState on a component that is not yet mounted. This is a no-op, but it might indicate a bug in your application. Instead, assign to this.state directly or define a state = {}; class property with the desired state in the SearchEngine component.

**Comment by benevbright** [comment]
Hi, @tomtom94 
Because ExtraHeader is not a child component of your screen component, you can't use `this.setState`. 
You need to communicate between your screen and ExtraHeader with using`navigation` prop, which is the same way you communicate with `react-navigation`'s header button.

So, instead `this.setState`, you need to use `navigation.setParams` and `naviagtion.getParam`.

I'll improve the example as you were going to try.

**Comment by tomtom94** [comment]
Thanks for your answer, it seems to be the solution ;)

However I tried something on your code example but it doesn't work yet (see below)

```
const ExtraHeader = ({ navigation }) => (
  <View
    style={{
      width: "100%",
      height: "100%",
      paddingHorizontal: 20,
      paddingVertical: 10
    }}
  >
    <View
      style={{
        backgroundColor: "white",
        flex: 1,
        borderRadius: 15,
        justifyContent: "center",
        alignItems: "center"
      }}
    >
      <Text
        style={{ color: "gray" }}
        onPress={() => {
          navigation.setParams({ ok: "ok" });
        }}
      >
        Search Here
      </Text>
    </View>
  </View>
);
```

error message : Undefined is not an object (evaluating 'navigation.setParams')

I like this module a lot but, do you have any clue ?

**Comment by benevbright** [comment]
Hi, @tomtom94 
I have to say to you I'm sorry. 
The module was passing wrong props to extraHeaderComponent. 
I fixed it and published 0.2.13.

And I updated example.
Please check [Expo Snack](https://snack.expo.io/@benevbright/react-navigation-collapsible) or [/example/src/ExtraHeaderScreen.js](https://github.com/benevbright/react-navigation-collapsible/tree/master/example/src/ExtraHeaderScreen.js)

**Comment by tomtom94** [comment]
It works like a charm, thanks a lot. This module is simple & awesome. 

Bye.
